### DISCUSSION TITLE
增加重新安装功能

### DISCUSSION THREAD

**Discussion by wang-qa** [1Panel-dev/1Panel#4222] [Announcements]
### 1Panel 版本

v1.10.1-lts

### 请描述您的需求或者改进建议

场景：定制化系统安装

1. 安装1panel ，可以正常启动
2. 升级系统后，系统会自动清除不在指定路径下的配置文件
3. 当前保留 安装在$HOME/1panel

预期：
可以手动复制 service等配置文件到指定路径，获取保留旧数据并重新安装（如端口号、安装路径），以便service等重新生成

### 请描述你建议的实现方案

生成配置好的 service，1pctl，直接使用软连接链接到指定路径下
并能正常启动

**Comment by wanghe-fit2cloud** [comment]
在操作系统升级之前，进入 `面板设置` 页面制作一个 1Panel 快照，然后用快照恢复就可以满足你的需求吧。

**Reply by wang-qa** [reply]
重启后service 无法启动，无法找到baseDIR
所以要重新链接或重新使用已有配置重新安装

**Reply by wanghe-fit2cloud** [reply]
嗯，这种情况也需要重新安装 1Panel，然后根据之前的快照恢复。

**Reply by wang-qa** [reply]
重新安装时，提示已安装，
如果选择先卸载，则会删除`卸载将会完全清除 1Panel 服务和数据目录，是否继续`
希望保数据，重新安装服务
目前是手动复制安装文件夹然后重新安装后，覆盖安装路径

**Reply by wanghe-fit2cloud** [reply]
没问题的，卸载 - 重新安装 - 同步快照 - 恢复，这个步骤会将数据目录恢复过来的。

> 我们也在后续版本考虑一下上述场景。

**Reply by wang-qa** [reply]
快照 应该不能 源和目标一致吧？会形成套娃
建议可以多个本地磁盘
![image](https://github.com/1Panel-dev/1Panel/assets/58168764/42905295-b242-496a-b1c0-265a3066bab2)
![image](https://github.com/1Panel-dev/1Panel/assets/58168764/a67d4bb1-6b3b-40c8-aecb-279bbede424d)

**Reply by wanghe-fit2cloud** [reply]
不会的，中间会有备份、恢复等临时目录的
